pkg/kubelet/kubelet.go

```diff
@@ -1731,6 +1731,8 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, podStatus *kubecont
 	}
 
 	apiPodStatus := kl.generateAPIPodStatus(pod, podStatus)
+	// The pod IP may be changed by api pod status if the pod is using host network. (See #24576)
```

Comment:
@pmorie I mean the pod ip will be set to host ip in `generateAPIPodStatus`. Sorry for replying so late, we have meetings a whole day. :)